src/utils/ceph.ts

```diff
@@ -152,3 +152,30 @@ export async function writeCephConf(clientName: string, cephConf: string, key: s
   await fsp.writeFile(keyringPath, keyring)
   await fsp.chmod(keyringPath, 0o600)
 }
+
+export async function sparsify(imageSpec: ImageSpec) {
+  console.log('Sparsify-ing ceph block device', imageSpec)
+  const startTime = new Date()
+  const {exitCode, stderr} = await execa('rbd', ['sparsify', '--no-progress', imageSpec], {
+    reject: false,
+    stdio: 'inherit',
+  })
+
+  const endTime = new Date()
+  const executionTime = endTime.getTime() - startTime.getTime()
+
+  console.log(`Sparsified ${imageSpec} in ${executionTime} ms`)
+
+  // 2 is "image does not exist" a.k.a ENOENT.
+  if (exitCode === 0 || exitCode === 2) {
+    return
+  }
+
+  // 30 is "read-only file system" a.k.a EROFS.
+  // This means that someone is using this block device and we cannot sparsify it right now.
+  if (exitCode === 30) {
+    return
+  }
```

Comment:
I think we should treat `30` as not completed, but also as not an error:

```suggestion
  // 2 is "image does not exist" a.k.a ENOENT.
  if (exitCode === 0 || exitCode === 2) {
    return true
  }

  // 30 is "read-only file system" a.k.a EROFS.
  // This means that someone is using this block device and we cannot sparsify it right now.
  if (exitCode === 30) {
    return false
  }
```

Basically I'm thinking of this scenario:

1. Volume begins trim because needs `needsTrim` is true and the volume is unmounted
2. Build starts so volume is mounted and `needsTrim` is set to true
3. Trim fails, but it sets `needsTrim` to false, so the volume is never trimmed

I think instead you can add a `.where('attachedToID', 'is', null)` condition in the API, so that we only _start_ trimming the volume when it's not in use, then if a build preempts the trim, it will pick back up after the volume is unused again.